git-link.el

```diff
@@ -164,6 +164,15 @@
   :type 'boolean
   :group 'git-link)
 
+(defcustom git-link-preferred-format '((lambda ()
+                                         (url-hexify-string (git-link--branch)))
+                                       (lambda ()
+                                         (url-hexify-string (git-link--tag)))
+                                       git-link--commit)
```

Comment:
What's the benefit to a list of functions vs just a symbol, e.g., `branch`, `tag`..? `git-link--branch` etc... are internal functions so caller should not use them.

Reply:
```el
(defcustom git-link-preferred-format '(branch tags commit))
; ...
(funcall (intern format "git-link--%s" style)))
```
```

Reply:
DONE.